scalding-serialization/src/main/scala/com/twitter/scalding/serialization/Boxed.scala

```diff
@@ -543,268 +543,540 @@ case class BoxedOrderedSerialization[K](box: K => Boxed[K],
+      (f212[K](_), classOf[Boxed212[K]]),
+      (f213[K](_), classOf[Boxed213[K]]),
+      (f214[K](_), classOf[Boxed214[K]]),
+      (f215[K](_), classOf[Boxed215[K]]),
+      (f216[K](_), classOf[Boxed216[K]]),
+      (f217[K](_), classOf[Boxed217[K]]),
+      (f218[K](_), classOf[Boxed218[K]]),
+      (f219[K](_), classOf[Boxed219[K]]),
+      (f220[K](_), classOf[Boxed220[K]]),
+      (f221[K](_), classOf[Boxed221[K]]),
+      (f222[K](_), classOf[Boxed222[K]]),
+      (f223[K](_), classOf[Boxed223[K]]),
+      (f224[K](_), classOf[Boxed224[K]]),
+      (f225[K](_), classOf[Boxed225[K]]),
+      (f226[K](_), classOf[Boxed226[K]]),
+      (f227[K](_), classOf[Boxed227[K]]),
+      (f228[K](_), classOf[Boxed228[K]]),
+      (f229[K](_), classOf[Boxed229[K]]),
+      (f230[K](_), classOf[Boxed230[K]]),
+      (f231[K](_), classOf[Boxed231[K]]),
+      (f232[K](_), classOf[Boxed232[K]]),
+      (f233[K](_), classOf[Boxed233[K]]),
+      (f234[K](_), classOf[Boxed234[K]]),
+      (f235[K](_), classOf[Boxed235[K]]),
+      (f236[K](_), classOf[Boxed236[K]]),
+      (f237[K](_), classOf[Boxed237[K]]),
+      (f238[K](_), classOf[Boxed238[K]]),
+      (f239[K](_), classOf[Boxed239[K]]),
+      (f240[K](_), classOf[Boxed240[K]]),
+      (f241[K](_), classOf[Boxed241[K]]),
+      (f242[K](_), classOf[Boxed242[K]]),
+      (f243[K](_), classOf[Boxed243[K]]),
+      (f244[K](_), classOf[Boxed244[K]]),
+      (f245[K](_), classOf[Boxed245[K]]),
+      (f246[K](_), classOf[Boxed246[K]]),
+      (f247[K](_), classOf[Boxed247[K]]),
+      (f248[K](_), classOf[Boxed248[K]]),
+      (f249[K](_), classOf[Boxed249[K]]),
+      (f250[K](_), classOf[Boxed250[K]]))
+
+  private[this] def boxes[K]: AtomicReference[List[(K => Boxed[K], Class[_ <: Boxed[K]])]] =
+    new AtomicReference(allBoxes[K])
+
+  def allClasses[K]: Seq[Class[_ <: Boxed[K]]] = allBoxes[K].map(_._2)
+
+  // The `Class` type is invariant in its type parameter, which is why this
+  // `_.isInstanceOf` is necessary and the `IsInstanceOf` warning is disabled.
+  //
+  // The `Throw` warning is suppressed because I (@Gabriel439) do not understand
+  // the rationale behind the `Boxed` mechanism enough to say whether or not
+  // this could be done in a more type-safe way.  However, I've added a more
+  // helpful error message in the interim.
+  @SuppressWarnings(Array("org.brianmckenna.wartremover.warts.IsInstanceOf", "org.brianmckenna.wartremover.warts.Throw"))
+  def next[K]: (K => Boxed[K], Class[Boxed[K]]) = boxes[K].get match {
     case list @ (h :: tail) if boxes.compareAndSet(list, tail) =>
       h.asInstanceOf[(K => Boxed[K], Class[Boxed[K]])]
     case (h :: tail) => next[K] // Try again
-    case Nil => sys.error("Exhausted the boxed classes")
+    case Nil => sys.error(
+      """|Scalding's ordered serialization logic exhausted the finite supply of boxed classes.
+           |
+           |Resolution: Report this error to the Scalding maintainers
```

Comment:
What about in the case where they actually need 251+ `Boxed` types and need a code change here?  Or is there some other mechanism that guarantees that will never happen